db/statistics.go

```diff
@@ -811,11 +811,13 @@ func WriteValidatorDepositWithdrawals(day uint64) error {
 	}
 
 	firstEpoch, lastEpoch := utils.GetFirstAndLastEpochForDay(day)
-	// for getting the withrawals / deposits for the current day we have to go 1 epoch in the past as they affect the balance one epoch after they have happend
-	if firstEpoch > 0 {
-		firstEpoch--
+
+	// for getting the withrawals / deposits for the current day we have to go 31 slots in the past as the balance for an epoch is the value after the first slot
```

Comment:
This is very counter-intuitive. Can you please add a bit more information to this comment please?

Reply:
I added more info, please have a look.